src/include/btree.i

```diff
@@ -981,8 +981,7 @@ __wt_page_can_evict(WT_SESSION_IMPL *session, WT_PAGE *page, int check_splits)
 	 * been written in the checkpoint, leaving the checkpoint inconsistent.
 	 */
 	if (btree->checkpointing &&
-	    (__wt_page_is_modified(page) ||
-	    F_ISSET(mod, WT_PM_REC_MULTIBLOCK))) {
+	    (__wt_page_is_modified(page) || F_ISSET(mod, WT_PM_REC_MASK))) {
```

Comment:
> if we don't bother maintaining a tree clean/dirty flag, we'll write checkpoints in every open tree on every checkpoint call. Maybe that won't cause a measurable performance problem, but I'd want evidence.

I'm not suggesting we not maintain a tree clean/dirty flag -- just suggesting that it might be OK if it was wrong, occasionally.